test/NuGet.Clients.Tests/NuGet.VisualStudio.Implementation.Test/Extensibility/NuGetProjectServiceTests.cs

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Moq;
+using NuGet.Configuration;
+using NuGet.PackageManagement.VisualStudio;
+using NuGet.PackageManagement.VisualStudio.Exceptions;
+using NuGet.ProjectManagement;
+using NuGet.ProjectManagement.Projects;
+using NuGet.VisualStudio.Contracts;
+using NuGet.VisualStudio.Implementation.Extensibility;
+using NuGet.VisualStudio.Telemetry;
+using Xunit;
+
+namespace NuGet.VisualStudio.Implementation.Test.Extensibility
+{
+    public class NuGetProjectServiceTests
+    {
+        [Fact]
+        public async Task GetInstalledPackagesAsync_CpsProjectNotNominated_ReturnsProjectNotReadyResult()
+        {
+            // Arrange
+            var projectGuid = Guid.NewGuid();
+
+            var settings = new Mock<ISettings>();
+            var telemetryProvider = new Mock<INuGetTelemetryProvider>(MockBehavior.Strict);
+
+            var project = new Mock<BuildIntegratedNuGetProject>();
+            project.Setup(p => p.GetPackageSpecsAndAdditionalMessagesAsync(It.IsAny<DependencyGraphCacheContext>()))
```

Comment:
Can you tell me how we're controlling nomination part here?
Can you give me comparable nomination works example?

Reply:
I don't see ProjectCache here.

Reply:
This unit test is testing `NuGetProjectService` only, everything else is mocked (I'm a bit of a purist when it comes to unit tests). There's no nomination, because this test is emulating the scenario where no nomination happened. There's no project cache because `NuGetProjectServices` doesn't use the project cache directly, hence the project cache is an implementation detail of classes not being used in this test.

Like all unit tests, this test is only as correct as how accurately the mock "simulates" the behaviour of the things it's mocking. In this case, if `CpsPackageReferenceProject` doesn't throw like this mock setup does, then this unit test is testing the wrong thing. But if the mock does behave like the thing it's mocking, then this unit test is shorter than having to create instances of every "real" class in the object hierarchy needed for the test, and the test is faster because it executes a lot less code.